lib/src/main/kotlin/starknet/data/types/transactions.kt

```diff
@@ -56,6 +58,26 @@ sealed class BlockHashOrTag() {
     abstract fun string(): String
 }
 
+class BlockHashOrTagSerializer(): KSerializer<BlockHashOrTag> {
+    override fun deserialize(decoder: Decoder): BlockHashOrTag {
+        val value = decoder.decodeString()
+
+        try {
+            val tag = BlockTag.valueOf(value)
+            return BlockHashOrTag.Tag(tag)
+        } catch (_: Exception) { }
+
+        return BlockHashOrTag.Hash(Felt.fromHex(value))
```

Comment:
I'm not a fan of using try catch like this, but I'm unsure how this could be improved.

Reply:
```
return try {
   val tag = BlockTag.valueOf(value)
   BlockHashOrTag.Tag(tag)
} catch (_: Exception) {
   BlockHashOrTag.Hash(Felt.fromHex(value))
}
```